cargo-creusot/src/setup/config.rs

```diff
@@ -0,0 +1,112 @@
+use serde::{Deserialize, Serialize};
+use std::{
+    fmt, fs,
+    path::{Path, PathBuf},
+};
+
+// identifies a version of the config file.
+// the goal is to avoid silently mis-interpreting a past or future version of
+// the config file whenever its format changes.
+pub const CURRENT_CONFIG_VERSION: i64 = 1;
+
+// bump CURRENT_CONFIG_VERSION if you change this definition
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "mode")]
+pub enum ToolsConfig {
+    #[serde(rename = "managed")]
+    Managed {
+        why3_path: PathBuf,
+        altergo_path: PathBuf,
+        z3: String,   // version
+        cvc4: String, // version
+        cvc5: String, // version
+    },
+    #[serde(rename = "external")]
+    External { why3_path: PathBuf, altergo_path: PathBuf },
+}
+
+// bump CURRENT_CONFIG_VERSION if you change this definition
+#[derive(Serialize, Deserialize)]
+pub struct Config {
+    pub tools: ToolsConfig,
+}
+
+pub enum Error {
+    NotFound,
+    Invalid(String),
+    WrongVersion(i64),
+}
+
+fn get_config_version(cfg: &toml::Value) -> Result<i64, String> {
+    cfg.get("version")
+        .ok_or("'version' field not found".to_string())?
+        .as_integer()
+        .ok_or("'version' is not an integer".to_string())
+}
+
+impl Config {
+    pub fn read_from_file(p: &Path) -> Result<Self, Error> {
+        if !p.is_file() {
+            return Err(Error::NotFound);
+        };
+        let s = match fs::read_to_string(p) {
+            Err(e) => return Err(Error::Invalid(e.to_string())),
+            Ok(s) => s,
+        };
```

Comment:
If we want more ergonomic error handling here we could look at `eyre` or more generally return a `Box<dyn Error>` object as the error case. Of you could define an instance of `From` for the appropriate types. 

It doesn't matter much here, but in case you wanted comments for future changes.

Reply:
Ah I see you're using `anyhow` down below.